src/Mod/Sketcher/Gui/DrawSketchController.h

```diff
@@ -305,12 +307,38 @@ class DrawSketchController
         }
     }
 
+    /** @brief Validates and unsets parameters that are below confusion threshold
+     * Only unsets parameters when user has finished editing to prevent
+     * degenerate geometry while allowing intermediate typing.
+     * Uses unit system to distinguish between angles (allow zero) and dimensions (validate zero) */
+    void validateAndUnsetFinishedParameters()
+    {
+        for (size_t i = 0; i < onViewParameters.size(); ++i) {

```

Comment:
A lot of the OVP loops use the old-school `for` construction because they need the index for various things. Since in this case you don't, I think it's a bit of a "code smell", and it would be better to use a range-based loop here.

Reply:
This way is safer indeed as it looks unlikely to introduce any issue. A better looking solution can be designed later if needed.